### DISCUSSION TITLE
Customizing Layouts with access top level tags (`html` and `body`) for Specific Pages in Next.js

### DISCUSSION THREAD

**Discussion by 0xdolan** [vercel/next.js#61226] [Help]
### Summary

Hello everyone!	

I have searched for information on nested layouts, but none of the resources addressed the specific concern of accessing top-level tags like `html` or `body` for a particular page.

Currently facing an issue in my `Next.js` project that utilizes `typescript` and `tailwindcss`. I would appreciate some guidance on customizing layouts for specific pages. Let me illustrate with an example:

I've implemented a `RootLayout` component that encompasses a `<NavBar />`, `<main>` content, and a `<Footer />` across the entire project. However, I'm encountering a requirement to establish distinct layouts for certain pages, such as 'Contact' and 'About,' which omit the `<NavBar />` and `<Footer />` components and feature unique backgrounds (images, videos, or colors).

Here's the existing `RootLayout` component from `app/layout.tsx`:

```jsx
export default function RootLayout({
  children}) {
  return (
    <html>
      <body>
        <NavBar />
        <main>{children}</main>
        <Footer />
      </body>
    </html>
  );
}
```

While attempting to create another `layout.tsx` file for a specific page, I've hit a roadblock. I can't directly access the `body` to modify the necessary components or alter the background color, image or video as full screen based on the requested page. For instance, if I want the 'About' page to exclude `<NavBar />` and `<Footer />` while having a red background color for the `body` of that page, it seems challenging. Using `usePathname` with a component like `<Footer />` doesn't seem ideal either, as it loads on all pages. Is it feasible to achieve this through nested layouts?

I'm using Next.js version 14.0.4, and I'd appreciate guidance on the best practices to handle this situation. Any help or examples would be fantastic!

Thank you for your time and assistance.

**Comment by makibat** [comment]
Hey!

Sounds like something [route groups](https://nextjs.org/docs/app/building-your-application/routing/route-groups) could solve. I suggest you also take a look at [this user's discussion](https://github.com/vercel/next.js/discussions/61110) which mentions a very similar issue.
